sunpy/net/fido_factory.py

```diff
@@ -41,68 +40,28 @@ class UnifiedResponse(Sequence):
     index the second dimension with ``::2``.
     """
 
-    def __init__(self, lst):
+    def __init__(self, *results):
         """
         Parameters
         ----------
-        lst : `object`
-            A single instance or an iterable of ``(QueryResponse, client)``
-            pairs or ``QueryResponse`` objects with a ``.client`` attribute.
+        results : `sunpy.net.base_client.BaseQueryResponse`
+            One or more QueryResponse objects.
         """
-
-        tmplst = []
-        # numfile is the number of files not the number of results.
+        self._list = []
         self._numfile = 0
-        if isinstance(lst, (QueryResponse, vsoQueryResponse)):
-            if not hasattr(lst, 'client'):
-                raise ValueError(
-                    ("A {} object is only a valid input to UnifiedResponse "
-                     "if it has a client attribute.").
-                    format(type(lst).__name__))
-            tmplst.append(lst)
-            self._numfile = len(lst)
-        else:
-            for block in lst:
-                if isinstance(block, tuple) and len(block) == 2:
-                    block[0].client = block[1]
-                    tmplst.append(block[0])
-                    self._numfile += len(block[0])
-                elif hasattr(block, 'client'):
-                    tmplst.append(block)
-                    self._numfile += len(block)
-                else:
-                    raise ValueError(
-                        "{} is not a valid input to UnifiedResponse.".format(type(lst)))
-        self._list = tmplst
+        for result in results:
+            if not isinstance(result, BaseQueryResponse):
```

Comment:
Would `issubclass` make more sense or does it change nothing?

Reply:
It should be passed instances (i.e. with results) rather than types here, so `isinstance` is really what we need.